daemons/fenced/fenced_commands.c

```diff
@@ -2748,17 +2748,22 @@ bool fencing_peer_active(crm_node_t *peer)
 
 void set_fencing_completed(remote_fencing_op_t * op)
 {
-#ifdef CLOCK_MONOTONIC
-    struct timespec tv;
+    bool have = false;
 
-    clock_gettime(CLOCK_MONOTONIC, &tv);
+#ifdef CLOCK_REALTIME
+    struct timespec tv;
 
-    op->completed = tv.tv_sec;
-    op->completed_nsec = tv.tv_nsec;
-#else
-    op->completed = time(NULL);
-    op->completed_nsec = 0L;
+    if (clock_gettime(CLOCK_REALTIME, &tv) == 0) {
```

Comment:
This is just used for history - right?
Which time does libqb-logging actually use? And is that configurable we might inherit it somehow.
Just to assure logs and history shown by crm_mon are in sync.

Reply:
Correct, just history. libqb uses:

    void
    qb_util_timespec_from_epoch_get(struct timespec *ts)
    {
    #ifdef CLOCK_REALTIME_COARSE
            clock_gettime(CLOCK_REALTIME_COARSE, ts);
    #else
            clock_gettime(CLOCK_REALTIME, ts);
    #endif
    }

CLOCK_REALTIME_COARSE is Linux- and architecture-specific, and is quicker but less precise. It has a fallback implementation using gettimeofday() if CLOCK_MONOTONIC is not defined, but no time() fallback. (Presumably that's sufficient for our purposes since libqb passes CI ...) This is public API available since at least our minimum dependency of 0.17, so I'll use it. FYI, there's no monotonic equivalent in libqb for getting a timespec.